### ISSUE TITLE
asbru for centos

### ISSUE THREAD

**Issue by johnshumon** [asbru-cm/asbru-cm#60]
Just wondering is there any installation package available for centos? current installation guide says it's available for debian/ubuntu, fedora, arch/manjaro only.

if not, is there a chance of centos package in near future?

**Comment by KlaasT** [comment]
We don't have any CentOS packages at the moment but you can try if one of the Fedora packages work for you.

**Comment by johnshumon** [comment]
```
curl -s https://packagecloud.io/install/repositories/asbru-cm/asbru-cm/script.rpm.sh | sudo bash
Detected operating system as centos/7.
Checking for curl...
Detected curl...
Downloading repository file: https://packagecloud.io/install/repositories/asbru-cm/asbru-cm/config_file.repo?os=centos&dist=7&source=script
done.
Installing pygpgme to verify GPG signatures...
Loaded plugins: fastestmirror, langpacks
asbru-cm_asbru-cm-source/signature                                                                                                                                                                                     |  836 B  00:00:00     
asbru-cm_asbru-cm-source/signature                                                                                                                                                                                     |  951 B  00:00:00 !!! 
Loading mirror speeds from cached hostfile
 * base: mirror.nsc.liu.se
 * epel: epel.mirror.far.fi
 * extras: mirror.nsc.liu.se
 * updates: mirror.nsc.liu.se
Package pygpgme-0.3-9.el7.x86_64 already installed and latest version
Nothing to do
Installing yum-utils...
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * base: mirror.nsc.liu.se
 * epel: epel.mirror.far.fi
 * extras: mirror.nsc.liu.se
 * updates: mirror.nsc.liu.se
Package yum-utils-1.1.31-42.el7.noarch already installed and latest version
Nothing to do
Generating yum cache for asbru-cm_asbru-cm...

The repository is setup! You can now install packages.
```

and then:
```
sudo yum install asbru-cm
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * base: mirror.nsc.liu.se
 * epel: epel.mirror.far.fi
 * extras: mirror.nsc.liu.se
 * updates: mirror.nsc.liu.se
No package asbru-cm available.
```

Any idea?

**Comment by KlaasT** [comment]
We are trying to build CentOS packages but that doesn't seem to be very easy. However in the mean time you can try to download RPMs directly at https://packagecloud.io/asbru-cm/asbru-cm and test them if they work. That would help us in the future if the dependencies might already work for CentOS.

**Comment by gfrenoy** [comment]
See issue #30 

We would really need help on dependencies from CentOS packagers.  If you can find time to contact them and ask how to get the perl-Gnome2-Vte package into CentOS, I'd believe Ásbrú will then be easy to deliver on that distribution.

**Comment by johnshumon** [comment]
@KlaasT That didn't work. 
```
sudo rpm -ivh asbru-cm-5.0.0-1.fc26.noarch.rpm 
[sudo] password for ashumon: 
error: Failed dependencies:
	perl(Crypt::Blowfish) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(Crypt::Rijndael) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(Expect) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(Gnome2::GConf) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(Gtk2::AppIndicator) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(Gtk2::Ex::Simple::List) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(Gtk2::Ex::Simple::TiedCommon) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(Gtk2::GladeXML) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(Gtk2::SourceView2) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(Gtk2::Unique) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(HTTP::Proxy) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(IO::Stty) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(IO::Tty) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(Net::ARP) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl(XML::Parser) is needed by asbru-cm-5.0.0-1.fc26.noarch
	perl-Gnome2-Vte is needed by asbru-cm-5.0.0-1.fc26.noarch
```

**Comment by johnshumon** [comment]
@gfrenoy `perl-Gnome2-Vte` package can be installed using `cpan`. I have managed to run `PAC Manager 4.5.5.7` which is also based on `perl-Gnome2-Vte`. Have a look at this: [script](https://github.com/telefax/Pac-manager-Dependency-installer-for-CentOS-7/blob/master/pmdic7.sh). All the necessary dependencies of perl are installed using cpan. This might give an idea how dependencies of asbru for centos can be sorted.

**Comment by gfrenoy** [comment]
So, as said in issue #30, this is an issue that needs to be addressed by RHEL7/CentOS packagers... 

Anyway, let's try to help with our limited CentOS knownledge :-/

_Disclaimer: I did not try any of these options, use it at your own risk !_

For the Perl & Gtk2 dependencies, don't you have to be bound to a "repository" that would give you those ? Something like the [EPEL repository](https://www.tecmint.com/how-to-enable-epel-repository-for-rhel-centos-6-5/) ?

And in any case, if you managed the dependencies by yourself, you can override the dependency check with the rpm option "--nodeps".  Or, better, get a copy of the git repository and start ```asbru-cm```.

And if you could find someone that has experience in making packages for CentOS, please contact him/her to help us :)

**Comment by Croydon** [comment]
A solution to this would be Flatpak support, see #83

**Comment by KlaasT** [comment]
@Croydon I don't really believe that Flatpak would help us here very much. We have many unsolved dependencies on CentOS which makes things problematic.

**Comment by popxunga** [comment]
Hi @gfrenoy 

I was able to get asbru-cm running on RHEL 7.5.
I'm a PAC user and I'll test asbru-cm it during the next days.

From a very brief test, everything seems mostly fine. 
Just one annoying glitch found so far: 
In order to show the asbru-cm main window, a double click on the systray icon is not enough; one click (left or right both are showing the context menu) plus another click in the "Show Window" option are required.

Some of the needed dependent packages are available from EPEL.
The dependent packages that are unavailable from EPEL needs to be built from source.

Most of the times the Fedora sources compiles fine on RHEL/CentOS.
I've just built the missing packages in my system from the Fedora 27 sources ( [https://apps.fedoraproject.org/packages/](url)). 

I'm not a packager nor a developer and I do not have much time to dedicate to this, but if someone is  interested, I can explain how-to build the missing packages.

Best regards.

**Comment by x-wolverine-x** [comment]
Hi, any news on here with CentOS 8? I was able to get it to work with CentOS 7, but no look with CentOS 8.

**Comment by KlaasT** [comment]
I will have a look at that. We are currently building Asbru on the bases of GTK3. Maybe that will bring us a step forward to CentOS 8.

**Comment by x-wolverine-x** [comment]
For CentOS 7 x86_64 users, you can try with the repo from Nux Dextop. http://li.nux.ro/download/nux/dextop/el7/x86_64/

**Comment by KlaasT** [comment]
The first CentOS 7/8 builds are done for our gtk3 branch. However we will probably need to fix quite some dependencies of the rpm package before it can be installed and run.

You will find it here. https://packagecloud.io/asbru-cm/gtk3/packages/el/8/asbru-cm-5.2.90~1570527220-1.el8.noarch.rpm

**Comment by x-wolverine-x** [comment]
Thx, I will wait for a ready to install version then.

**Comment by popxunga** [comment]
Hi ...
I've just built the missing el7 packages from source as the latest asbru-cm-6.0.0~1570889814-1.el7.src.rpm as well using the following source packages:

perl-ExtUtils-Manifest-1.72-439.el7.src.rpm
perl-Gtk3-SimpleList-0.18-4.el7.src.rpm
perl-ExtUtils-MakeMaker-6.98-1.el7.src.rpm
perl-CPAN-Meta-2.150010-439.el7.src.rpm
perl-JSON-PP-4.04-2.el7.src.rpm
perl-CPAN-Meta-YAML-0.018-440.el7.src.rpm

All them can be found [here](https://apps.fedoraproject.org/packages/) except the perl-ExtUtils-MakeMaker-6.98-1.el7.src.rpm that I've found [here](https://centos.pkgs.org/7/openfusion-x86_64/perl-ExtUtils-MakeMaker-6.98-1.of.el7.noarch.rpm.html).

Briefly tested asbru 6 and found no issues so far.

Let me know If I can help somehow.
Regards

**Comment by KlaasT** [comment]
Thanks for your work. I think the goal should be that the end user wouldn't have to build any packages. Maybe we can resolve these dependencies from the OS repositories.

**Comment by popxunga** [comment]
Of course that's the goal but considering the el7 platform that is now at
least 3 years old, the reality is that the dependent packages required for
pac 4.5.x, asbru-cm 5.x or asbru-cm 6.x are still nowhere.

The EPEL 7 should/could be the repo where to make them available.

Another possible (?) alternative would be store them in the same repo from
where asbru-cm is available now, but I don't know if this is against any
rules.

A domingo, 13/10/2019, 16:06, Klaas Tammling <notifications@github.com>
escreveu:

**Comment by KlaasT** [comment]
I don't know how complicated it is to get new RPMs into the EPEL repositories. I agree that the best place for the additional packages would be EPEL.

We could upload them to our own repositories however that mean even more maintenance.

With adding EPEL to my local CentOS installation I was able to narrow missing dependencies down to Crypt::Blowfish, Gtk3::SimpleList and HTTP::Proxy. For the last I don't even think we will need that one. I cannot find any references to that in the code.

**Comment by x-wolverine-x** [comment]
@KlaasT you know when there will be correct centos (el) 8 rpm? All available are "not an rpm package (or package manifest)" 
THX

**Comment by x-wolverine-x** [comment]
I think the chance to get it to the EPEL Repo is close to zero. But maybe to rpmfusion.

**Comment by hanspr** [comment]
I don't know anything about creating packages, but I supouse you can execute bash commands during any installation process. So if you can add this lines to your install scripts, you will be able to install any dependency on any OS that has some missing package library

1. For centos add dependency to : perl-App-cpanminus
2. execute : cpanm -iqn Crypt::Blowfish
3. execute : cpanm -iqn Gtk3::SimpleList
4. execute : cpanm -iqn HTTP::Proxy

With this approach you can install any missing package directly from cpan into any OS, centos 7,6
 Hope it helps

**Comment by x-wolverine-x** [comment]
That's true, for but for CentOS 6 and 7 I see no Problem to get it work, there are many ways to make it. But for CentOS 8 it's impossible for now for me 👎  If you have any helpful tips I would be very thankful.

**Comment by x-wolverine-x** [comment]
After a fresh install from CentOS 8 with enabled EPEL Repo.
Try to install from the gtk3 repo
![image](https://user-images.githubusercontent.com/21344031/68034160-7991d980-fcc1-11e9-8e89-bd11e26823d4.png)

**Comment by x-wolverine-x** [comment]
And there is no perl-Gtk3 or vte package for CentOS 8 / EPEL 8 for now.

**Comment by hanspr** [comment]
I think the problems with missing packages should work with the same principle as long as this package is available from the start : perl-App-cpanminus

If it does, then you are set to install all the missing ones. By executing the commands

*cpanm -iqn Crypt::Blowfish
*cpanm -iqn Crypt::Rijindael
*cpanm -iqn Glib::Object::Introspection
*cpanm -iqn Gtk3::SimpleList
...
...

Now the vte, lets see if it runs with that version, I think there should be no problems to run, if the vte shipped with Centos is old, then the user will have to find a way with Centos to request for them to upgrade to a newer one. Or live with the limitation provided by the published by Centos.

**Comment by hanspr** [comment]
And well, I may be stating the obvious, but that package should not have any required dependencies to any of the missing ones so it does not refuses to install.

**Comment by x-wolverine-x** [comment]
@hanspr first, thank you for your time and help. :-)
How to install the perl-Gtk3 in CentOS 8? Because I need this to install Gtk3::SimpleList for example.

**Comment by hanspr** [comment]
If it can not be installed with

cpanm -iqn Gtk3

then we are out of luck, game over.

Let me install a VM with centos 8 to test a manual install of asbru and I tell you if it is or not possible

**Comment by hanspr** [comment]
I'm downloading Centos 8,

meanwhile I will try to explain my self just in case.

There is the distro package manager for each linux distribution. yum, rpm, apt, etc.

And there is cpanm. That you could say is the package manager for perl libraries (similar to pip in python if you are familiar with this).

You can install perl libraries using the distro package manager. But if a particular package does not exists in that distro, you still can install "perl libraries only", using cpanm. The repository for perl libraries is OS agnostic, so it should normally work for any Distro.

But, what could happen is if the perl Library has "C" bindings to a particular C library, that C library has to be installed first.

So if Centos 8 has "no gtk3 C libraries", again cpanm -iqn Gtk3, will not work because it will not find the C libraries to bind to.

So the logic would be, to create a package that requires the C libraries that are need it. And then install all the perl libraries that do not have a distro package already available, using `cpanm`

I hope, I explained my self better, the download is almost done, let me see what Centos 8 has or does not have available.

**Comment by x-wolverine-x** [comment]
Got your point. I hope that you can find a way to make it.

**Comment by hanspr** [comment]
Right now I'm going for the third install attempt, it crashes on the VM at the end and it does not install on VirutalBox.

**Comment by x-wolverine-x** [comment]
You use the 8 or Stream ISO?

**Comment by hanspr** [comment]
CentOS-8-x86_64-1905-dvd1.iso

It always crashes at the end when asks to accept a license,

**Comment by x-wolverine-x** [comment]
I will try to install it in VirtualBox now

**Comment by hanspr** [comment]
I always arrive to this final screen

![imagen](https://user-images.githubusercontent.com/1572396/68054686-780de480-fcb4-11e9-84d2-b8f3a94db810.png)

I will install a minimum and then install everything else with the package manager

**Comment by x-wolverine-x** [comment]
I install Server with GUI for testing.

**Comment by x-wolverine-x** [comment]
Is working fine... 
![image](https://user-images.githubusercontent.com/21344031/68055613-6139b300-fcf1-11e9-8b34-d4da26a9a72e.png)

**Comment by hanspr** [comment]
Since I upgraded to mint 19, I've been having all sort of weird problems. I will have to solve this one to be able to move on. Could you share, your virutal machine settings just in case I'm choosing the wrong ones.

**Comment by hanspr** [comment]
Done, it failed because I was selecting spanish at the begining, it works if you install in english, so I guess the Centos team thinks that we all live in New York

**Comment by x-wolverine-x** [comment]
I always install in English, and just change the keyboard

**Comment by hanspr** [comment]
@x-wolverine-x 

I'm really sorry, I hit a wall. The problem is with library OSSP::uuid

We should change that dependency, is a non standard library from CPAN, I think that we will have more problems as years pass if we keep this dependency in the code.

I will see how difficult is to move away from this library and use a more standard one.

**Comment by KlaasT** [comment]
I hope at the weekend I will find time to rewrite some of our build environment. So we can build Red Hat/CentOS independendly from a Fedora or opensuse

**Comment by hanspr** [comment]
Hi,

I managed to substitute OSSP for Data::uuid, but that did not help.

When attempting to build gtk3, there are missing "devel" c library packages. So you can not use cpanm because this procedure what it does is to compile the c bindings based on those files. For example I could not find package : gboject-introspection-devel, thou gboject-introspection (binary) exists and is installed by default. But without the devel files, you can not install perl-gtk3.

And that is the main reason why you see so many missing packages available. The process is moving slow.

For what I've been reading, is to early to jump into Centos 8.

We could try to keep fighting our way in, but every time we would have to build an installer that would rely on downloading and compiling every C and perl Library needed.

I would wait a couple of months and monitor how is the availability of packages going and when the missing number is not to high, then we could use a mixed approach as I was suggesting.

**Comment by x-wolverine-x** [comment]
That was the same that I see... every missing dependency kick me to many other missing.

**Comment by x-wolverine-x** [comment]
Maybe someone will care for this https://bugzilla.redhat.com/show_bug.cgi?id=1759801
Maybe we have to wait for EPEL 8.1 or EPEL 8.2

**Comment by popxunga** [comment]
Hi @gfrenoy 

As per your request on #269:

I start doing this while I was a user of pac 3 or 4 and with EL6. 
The life was easier with EL6 because lots of the required packages were available from EPEL 6. 
At that time I had to compile from source just a few packages.
Once EL7 came out things got a lot more complicated.

1 - First you need to install the basic development tools on your CentOS 7 (gcc, etc,).
2 - Install and enable the EPEL 7 repository.
3 - Then to build one of the missing packages
     a) grab its src.rpm from koji 
     b) Logged in as your user, install with: rpm -i <package_name>.src.rpm ( a structure of folders is created under ~/rpmbuild.
     c) Navigate to ~/rpmbuild/SPECS 
     d) Ensure you have everything needed to build the package by executing:
         sudo yum-builddep <package_name>.src.rpm
     e) if something required is missing on d) then build it first.
     f) Once you get everything required to build one package, build it with:
         rpmbuild -ba <package_name>.src.rpm
         The resulting rpm's will be written under ~/rpmbuild/RPMS .
         And the src.rpm's under ~/rpmbuild/SRPMS

4) Then build a local repository by:
    a) creating a folder let's say in ~/localrepo/packages
    b) copy the built rpm's into ~/localrepo/packages
    c) change to ~/localrepo and execute: "createrepo ." (this will create a folder repodata).
    d) finally let your system know that local repo exists by creating this:

[~]$ cat /etc/yum.repos.d/local_repo.repo
[local_repo]
name = My Local Repository
failovermethod = priority
enabled=1
baseurl=file:///your_home_dir/localrepo
gpgcheck = 0

5) sudo yum clean all ; sudo yum update  

Note:  Each time you add a new package into ~/localrepo/packages (step 4b), remove the repodata folder and repeat the create "createrepo ." command as stated in step 4c) and then run step 5 again.

If you prefer avoiding all this mess, I can share the packages I've built so far. Let me know which ones you need to pac 5.x or 6.x.

Regards

**Comment by gfrenoy** [comment]
Yes if you already have the packages built, that'd be great if you can share them so that we can host them on packagecloud.io ...

**Comment by KlaasT** [comment]
I just admit I didn't manage to do any work on CentOS and the build system last week. (We had some big trouble at work). There will be some updates to this issue soon.

Any packages you can share and maybe the .spec files will be great.

**Comment by popxunga** [comment]
@gfrenoy 
Please let me know where to transfer the files.
Regards

**Comment by x-wolverine-x** [comment]
@popxunga are you able to do this for the CentOS / EPEL 8 also?

**Comment by popxunga** [comment]
Hi...
I guess that some of the required packages doesn't exist on EPEL 8 yet. But
I'll have a look.
Regards.

A quarta, 13/11/2019, 05:25, x-wolverine-x <notifications@github.com>
escreveu:

**Comment by popxunga** [comment]
@x-wolverine-x @gfrenoy @KlaasT 

Hi guys ...

Think I may have good news for CentOS 8 as well:

![image](https://user-images.githubusercontent.com/37672088/68792735-e2733d00-0643-11ea-8b36-fa72f65f4884.png)

Regards.

**Comment by x-wolverine-x** [comment]
@popxunga you made my day. How you mage this for CentOS 8? I'm still try to find the right src rpm to build my own rpm for testing. xD

**Comment by x-wolverine-x** [comment]
@popxunga is it possible to get the CentOS 8 rpm's ? Or any hint what src rpm's to use?

**Comment by popxunga** [comment]
Hi @x-wolverine-x ...
I've asked already where to upload the files but as you can see I got no answer so far.

**Comment by KlaasT** [comment]
How big are the packages? For now you could send them to klaas@tammling.hamburg and I would have a look at them and upload them to our repository.

**Comment by popxunga** [comment]
Small stuff. Sent you and invitation from github ... Have you got it ?

**Comment by x-wolverine-x** [comment]
@popxunga maybe you can send me them to x.wolverine.x@hotmail.de too?
@KlaasT if it don't work to put them to your repo for some reason, I can setup a repo for them if needed.

**Comment by popxunga** [comment]
Hi @x-wolverine-x ... Sent !

**Comment by KlaasT** [comment]
I received the tar.gz just now. Give me a few minutes and I will upload them. For some reason I didn't get any invitation from Github.

**Comment by popxunga** [comment]
Note that those packages installation  should/need to be tested before
release them to public.
Better upload them to a test repository case you have it.
I was setting up a repository myself on GitHub, but I'm not sure that this
kind of usage (repositories for binary packages distribution) is allowed
there. Maybe because the repo was created as private or because meanwhile I
cancelled the invitation once you gave me your email contact, you didn't
got the invitation.
Best regards.


A quinta, 14/11/2019, 19:16, Klaas Tammling <notifications@github.com>
escreveu:

**Comment by KlaasT** [comment]
Okay.  I uploaded the EL 7 packages into a test repository.

curl -s https://packagecloud.io/install/repositories/asbru-cm/el-test/script.rpm.sh | sudo bash

**Comment by x-wolverine-x** [comment]
@popxunga for CentOS8 there are some missing
![image](https://user-images.githubusercontent.com/21344031/68892780-03ab5a80-0724-11ea-9939-b29d371fe3ac.png)

**Comment by KlaasT** [comment]
Will the EL 7 work for EL 8? Then I  would copy them over.

**Comment by x-wolverine-x** [comment]
@KlaasT is it possible to use vte291 instant of vte?

**Comment by x-wolverine-x** [comment]
> Will the EL 7 work for EL 8? Then I would copy them over.
I will try

**Comment by popxunga** [comment]
No.

A quinta, 14/11/2019, 20:18, Klaas Tammling <notifications@github.com>
escreveu:

**Comment by KlaasT** [comment]
Okay I found the missing emails. I will try to get them out of the spam quarantine.

**Comment by popxunga** [comment]
I'm sure I built them. Double check in the el8.tar.gz if the files are
there.

A quinta, 14/11/2019, 20:16, x-wolverine-x <notifications@github.com>
escreveu:

**Comment by x-wolverine-x** [comment]
Sorry my fault of course... xD I don't checked the noarch... sorry.
![image](https://user-images.githubusercontent.com/21344031/68893409-5c2f2780-0725-11ea-842b-66bac10fc27e.png)

**Comment by popxunga** [comment]
You have the noarch and x86_64 folders.

A quinta, 14/11/2019, 20:21, Luís Pontes <popxunga@gmail.com> escreveu:

**Comment by KlaasT** [comment]
EL8 has been uploaded.

**Comment by x-wolverine-x** [comment]
![image](https://user-images.githubusercontent.com/21344031/68893474-854fb800-0725-11ea-8e72-16e20fb60819.png)

A big thank you @popxunga

**Comment by popxunga** [comment]
I sent you 2 emails.
One for el7 and the other for el8.
You should put the el7 files in a el7 repo
And the el8 files in a el8 repo.
Please do not mix el7 with el8 stuff.


A quinta, 14/11/2019, 20:20, Luís Pontes <popxunga@gmail.com> escreveu:

**Comment by x-wolverine-x** [comment]
Repos are fine and not mixed up, it just was all up to my blindness.
BIG thank you guys for make this happen, you really make me very happy.

**Comment by popxunga** [comment]
**Note:**
The EL7files I've sent you support both, Asbru CM 5.2.x. and Asbru CM 6.x 
The EL8 files supports Asbru CM 6.x only.

**Comment by KlaasT** [comment]
I'm just waiting for some more feedback from CentOS or Red Hat users. If everything works alright so far I'm considering to push the packages to the main repository to make it easier for users to install the software and resolve dependencies.

**Comment by x-wolverine-x** [comment]
@KlaasT Test it on many CentOS 8 installations, and all working with the test-repo.

**Comment by gfrenoy** [comment]
By doing the follow, I was able to run Ásbrú on my CentOS 7 live VM:

```
sudo yum install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
curl -s https://packagecloud.io/install/repositories/asbru-cm/el-test/script.rpm.sh | sudo bash
curl -s https://packagecloud.io/install/repositories/asbru-cm/gtk3/script.rpm.sh | sudo bash
sudo yum install asbru-cm
```

It is working fine :-)  I just noticed an error message when starting a terminal:

```
*** This build of Glib was compiled with glib 2.56.1, but is currently running with 2.50.3, which is too old.  We'll continue, but expect problems!
```

I'd guess it's the vte built that was not done with the same Glib version as the target environment.  Is there anything we can do anything about this ?

**Comment by popxunga** [comment]
The packages were built on a CentOS 7 virtual machine which I had to drop due to space constraints. Did you fully updated your Centos 7 VM before doing the test ?
My RHEL 7.7. host has glib2 2.56.1. I would expect that your CentOS VM has the same version.

**Comment by gfrenoy** [comment]
> Did you fully updated your Centos 7 VM before doing the test ?

Good point !  No, I did not... And as it's only a "live" VM (no hard disk), this operation will typically fails so let's assume it's OK.  Thanks for the quick feedback.

**Comment by popxunga** [comment]
Hi @gfrenoy & @KlaasT 
Is there any interest to get Asbru 5.2.x supported on el8 as well ?

**Comment by KlaasT** [comment]
Currently we are trying to bring all effort into Asbru 6.x and rather prefer not to get busy in porting 5.2.x to new distributions.

**Comment by aTanCS** [comment]
Hi, I've just installed Asbru on Centos 7 an got this error:

```
Typelib file for namespace 'Wnck', version '3.0' not found at /usr/lib64/perl5/vendor_perl/Glib/Object/Introspection.pm line 108.
BEGIN failed--compilation aborted at /usr/share/asbru-cm/lib/PACUtils.pm line 54.
Compilation failed in require at /usr/share/asbru-cm/lib/PACMain.pm line 53.
BEGIN failed--compilation aborted at /usr/share/asbru-cm/lib/PACMain.pm line 53.
Compilation failed in require at /bin/asbru-cm line 92.
BEGIN failed--compilation aborted at /bin/asbru-cm line 92.
```

I guess it is related to #202 , but I can't find gir1.2-wnck-3.0 for Centos 7.

**Comment by aTanCS** [comment]
> I guess it is related to #202 , but I can't find gir1.2-wnck-3.0 for Centos 7.

This fixed all missing dependencies:
```
yum install libwnck3 vte291
```

**Comment by gfrenoy** [comment]
@aTanCS Thanks for your feedback.

**Comment by KlaasT** [comment]
@aTanCS Just wanted to ask back if this issue has been fixed yet. Did you try an installation using our snapshots repository? The dependency should be resolved there. Thanks.

**Comment by KlaasT** [comment]
Closed #60
